react-file-uploads/server/server.ts

```diff
@@ -0,0 +1,79 @@
+import 'dotenv/config';
+import express from 'express';
+import pg from 'pg';
+import {
+  ClientError,
+  errorMiddleware,
+  uploadsMiddleware,
+} from './lib/index.js';
+
+type Image = {
+  imageId: number;
+  url: string;
+  caption: string;
+};
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.static('public'));
+app.use(express.json());
+
+app.post(
+  '/api/uploads',
+  uploadsMiddleware.single('image'),
+  async (req, res, next) => {
+    try {
+      if (!req.file) throw new ClientError(400, 'no file field in request');
+      const { caption } = req.body as Partial<Image>;
+      if (!caption) {
+        throw new ClientError(400, 'caption is a required field');
+      }
+      const imageUrl = '/images/' + req.file.filename;
+      if (!imageUrl) throw new ClientError(400, 'url is a required field');
+      console.log(imageUrl);
+      const [url] = imageUrl;
```

Comment:
I'm confused by this statement. It says to destructure the array `imageUrl` and assign the value of the first element in the array to the variable `url`. But `imageUrl` is a string, not an array.

Thanks to JS magic (dark magic in this case), since `imageUrl` is a string, it will be destructured as if it were an array of the letters in the string. That means that `url` will take on the value of the first letter in `imagUrl`, which will be `/`.

I think you just want to pass `imageUrl` directly in the params.